Export appropriate metadata to Koji for flatpaks

owtaylor opened the pull request:
In the extra data we send to Koji for a Flatpak build, include:

```    
'flatpak': True,
'source_modules': [<specified module for the build>],
'modules': [<resolved list of module name-stream-version>]
```
    
This is sufficient information to know when a rebuild needs to be triggered and to start a rebuild.

vrutkovs commented on atomic_reactor/plugins/pre_resolve_module_compose.py:
```diff
@@ -206,13 +216,14 @@ def _resolve_compose(self):
-            else:
-                source = self.module_name + '-' + self.module_stream
+        if self.module_version is not None:
+            source_spec = self.module_name + '-' + self.module_stream + '-' + self.module_version
```
What about:
```
source_spec = "{0}-{1}".format(self.module_name, self.module_stream)
if self.module_version:
    source_spec = "{0}-{1}".format(source_spec, self.module_version)
```

This would avoid concatenation and convert into strings properly

owtaylor commented:
I think string concatenation is pretty optimized, probably as fast as getting the format machinery involved for this, and the arguments are supposed to be strings, but sure, I've switched it to:

```
fmt = '{n}-{s}' if self.module_version is None else '{n}-{s}-{v}'
source_spec = fmt.format(n=self.module_name, s=self.module_stream, v=self.module_version)
```

(optimizing for readability.)

Also added information to koji.md.

vrutkovs commented:
Needs a rebase to resolve conflicts

owtaylor commented:
Rebased
